include/oneapi/dpl/pstl/utils.h

```diff
@@ -784,6 +784,29 @@ union __lazy_ctor_storage
     }
 };
 
+// Utility that returns the smallest type size in a tuple.
+template <typename _Tuple>
+class __min_tuple_type_size;
+
+template <typename _T>
+class __min_tuple_type_size<std::tuple<_T>>
+{
+  public:
+    static constexpr std::size_t value = sizeof(_T);
+};
+
+template <typename _T, typename... _Ts>
+class __min_tuple_type_size<std::tuple<_T, _Ts...>>
+{
+    static constexpr std::size_t __min_type_value_ts = __min_tuple_type_size<std::tuple<_Ts...>>::value;
+
+  public:
+    static constexpr std::size_t value = std::min(sizeof(_T), __min_type_value_ts);
+};
+
+template <typename _Tuple>
+inline constexpr std::size_t __min_tuple_type_size_v = __min_tuple_type_size<_Tuple>::value;
```

Comment:
This can be simplified:

```
template<typename Tuple>
struct __min_tuple_type_size;

template<typename... Ts>
struct __min_tuple_type_size<std::tuple<Ts...>> {
    static constexpr std::size_t value = std::min({sizeof(Ts)...});
};
```

`_v` alias is not-necessary as it is used only once.

Reply:
Thanks, I have replaced this `__min_tuple_type_size` with `__min_nested_type_size` which avoids having to flatten the tuple first and applied these ideas there.